### ISSUE TITLE
traits with static vars

### ISSUE THREAD

**Issue by dominikzogg** [facebook/hhvm#3826]
In php this example works, cause it assigns the state of the trait at the moment of the use statement within the class. In hhvm it doesn't.

``` .php
<?php

trait SampleTrait
{
    /**
     * @var string
     */
    private static $sample;

    final public static function setSample($sample)
    {
        self::$sample = $sample;
    }

    final public static function getSample()
    {
        return self::$sample;
    }
}

SampleTrait::setSample('test');

class SampleClass
{
    use SampleTrait;
}

$sample = new SampleClass();
var_dump($sample::getSample());
```

php output: 'test';
hhvm 3.3 output: null

Its an edgecase i need in a accessor lib: https://github.com/saxulum/saxulum-accessor

**Comment by dshafik** [comment]
This issue appears to be the same behavior I've logged in the PHP.net bug tracker [here](https://bugs.php.net/bug.php?id=70437).

> If you have a trait that defines a static property, and you modify said property from within a trait inherited method, the modifications are done to a trait instance of the property, as opposed to the inherited-from-trait static property within the class.

Small reproducing script can be found [here](https://3v4l.org/UZ6gq)

**Comment by lexidor** [comment]
> Hmm, this is odd behavior http://3v4l.org/EShiW

Yes it is odd. Your 3v4l shows that the PHP project decided this was so crazy, that they are going to ban it.
_Deprecated: Calling static trait method SampleTrait::setSample is deprecated, it should only be called on a class using the trait in /in/EShiW on line 21_

Whatever hhvm does will make sense for hhvm. HHVM does not attempt to be bug for bug compatible with PHP since the release of 4.0.0.
